nflow-engine/src/main/java/io/nflow/engine/internal/workflow/StateExecutionImpl.java

```diff
@@ -103,18 +104,16 @@ public String getVariable(String name) {
   @Override
   @SuppressWarnings("unchecked")
   public <T> T getVariable(String name, Class<T> type, T defaultValue) {
-    if (instance.stateVariables.containsKey(name)) {
-      return (T) objectMapper.convertToObject(type, name, instance.stateVariables.get(name));
+    String value = instance.stateVariables.get(name);
```

Comment:
Does this change the semantincs?
Previously an explicit null value in db would not return `defaultValue` but `null` form this method.
I'm not against the semantics change, but maybe we should at least document it in changes.

Reply:
If stateVariable value is null, objectMapper.convertToObject throws NPE, so it does not change anything that was working before. Instead it fixes the case when null values has been stored for a state variable, and uses the default value instead as I would expect.